### DISCUSSION TITLE
working with the Statistics in the ColumnChunkMetaData object

### DISCUSSION THREAD

**Discussion by lnimong** [G-Research/ParquetSharp#335] [Q&A] [answered]
Hello 

THE CONTEXT
------------------

I'm using ParquetSharp to create a lot of files (1M or 2M rows with groups of around 100K rows)

I'm trying to make a "filter" feature 
Basically the idea is to find every rows having of a specific value on a specific column
I use the Statistics in the ColumnChunkMetaData object to avoid reading useless groups (if the value I'm looking for is not in range [max, min] of my column in the group, I avoid reading the group)

```
columnStatistics =>
            {
                try
                {
                    if (columnStatistics.Min is not null && targetValue.CompareTo(columnStatistics.Min) < 0)
                        return false;  //=> SKIP THE GROUP

                    if (columnStatistics.Max is not null && targetValue.CompareTo(columnStatistics.Max) > 0)
                        return false; //=> SKIP THE GROUP
                }
                catch
                {
                    return false; //=> SKIP THE GROUP because there is an exception (mostly Casting in the type I need)
                }

                return true;
            
```

It works fine with types such as int double...,
BUT, not for the string type
The reason is that the data I find in the statistics object (Min and Max value) are not of type string But they're ParquetSharp.ByteArray
Even tho that column was written using a LogicalColumnWriter<String>

THE QUESTION
---------------------
I do understand that we might ton want to always load a bunch of data  into that statistics object that could take an unknown memory space...
BUT, 
... Is there any way to access the Min an Max values of a column of string items ?  

```
ColumnChunkMetaData.Statistics.MinUntyped // not a string on a column written with a LogicalColumnWriter<string>
ColumnChunkMetaData.Statistics.MaxUntyped // not a string on a column written with a LogicalColumnWriter<string>
```

Or if you have any idea to better implement what I'm trying to achieve it would of course be welcome :)

**Comment by lnimong** [comment, marked as answer]
found My answer
I just didn"t know how to wotk with IntPtr 
Marshal.PtrToStringUTF8
did the trick

**Reply by adamreeve** [reply]
There's also `LogicalRead.ToString` in ParquetSharp that can be used for this purpose
